queue_job/models/queue_job.py

```diff
@@ -506,3 +509,31 @@ def _test_job(self, failure_rate=0):
         _logger.info("Running test job.")
         if random.random() <= failure_rate:
             raise JobError("Job failed")
+
+    def _call_webhook(self, **kwargs):
```

Comment:
I don't understand why we have to handle this here. If you  provide your own handler you can do whatever you want, including calling a webhook. I'd drop this part.